src/io/geotiff.rs

```diff
@@ -21,6 +21,25 @@ impl<R: Read + Seek> CogReader<R> {
         Ok(Self { decoder })
     }
 
+    /// Decode GeoTIFF image to an [`ndarray::Array`]
+    fn ndarray(&mut self) -> TiffResult<Array2<f32>> {
+        // Get image dimensions
+        let (width, height): (u32, u32) = self.decoder.dimensions()?;
+
+        // Get image pixel data
+        let decode_result = self.decoder.read_image()?;
+        let image_data: Vec<f32> = match decode_result {
+            DecodingResult::F32(img_data) => img_data,
+            _ => unimplemented!("Data types other than float32 are not yet supported."),
+        };
+
+        // Put image pixel data into an ndarray
+        let vec_data = Array2::from_shape_vec((height as usize, width as usize), image_data)
+            .map_err(|_| TiffFormatError::InvalidDimensions(height, width))?;
+
+        Ok(vec_data)
+    }
```

Comment:
Was playing around with wrapping parts of this in the [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) trait, but got caught up fighting the borrow checker when trying to use it in another part of the crate, so will leave things here for now.